pageserver/src/tenant/vectored_blob_io.rs

```diff
@@ -50,6 +51,31 @@ pub struct VectoredBlobsBuf {
     pub blobs: Vec<VectoredBlob>,
 }
 
+/// Decompresses the buffer if needed.
+///
+/// Returns an owned buffer if decompression is performed, otherwise return `None`.
+pub(crate) async fn decompress(
```

Comment:
Have you considered making this a method on `VectoredBlob`? 
Then we might be able to make the `compression_bits` field private, which would be nice (and avoid the newtype I asked for :) )

Reply:
done in 32284e58fc816ba5554a62e7bf5132695bf9478b